soccer/optimization/Gradient1DConfig.hpp

```diff
@@ -71,8 +68,7 @@ class Gradient1DConfig {
           maxValue(maxValue),
           maxThresh(maxThresh) {}
 
-    std::tuple<float, float>(*f)(float, FunctionArgs*);
-    FunctionArgs* args;
+    std::function<std::tuple<float, float>(float)>* f;
```

Comment:
Um, I don't think this needs to be pointer. Right now its a pointer to a function pointer I think.